### PR TITLE
Don't unpickle schema in I/O process

### PR DESCRIPTION
**Pull request by fantix**
- [x] Add pickled user schema
- [x] Add pickled global schema
- [x] Use pickled user/global schema in compiler client
- [x] Move inteprete_backend_error back to the compiler
- [x] Move parsing of JSON schema to the compiler
- [x] Allow compiler initialization with JSON schema
- [x] Edit introspect_user/global_schema
- [x] Deal with state serializer
- [x] Drop unpickled schema in dbview

### CODE REVIEW

File: edb/graphql/extension.pyx [hunk 1]
```diff
@@ -186,14 +186,12 @@ async def handle_request(
...
             if static_exc is errormech.SchemaRequired:
-                ex = errormech.interpret_backend_error(
-                    s_schema.ChainedSchema(
-                        tenant.server.get_std_schema(),
-                        db.user_schema,
-                        tenant.get_global_schema(),
-                    ),
+                compiler_pool = tenant.server.get_compiler_pool()
+                ex = await compiler_pool.interpret_backend_error(
+                    db.user_schema_pickled,
```
Comment (elprans) [inline edb/graphql/extension.pyx:191]: It's either a "pickled user schema" or "user schema pickle", I prefer the latter.

```suggestion
                    db.user_schema_pickle,
```

File: edb/server/compiler/compiler.py [hunk 2]
```diff
@@ -387,6 +387,8 @@ class CompilerState:
 
     @functools.cached_property
     def state_serializer_factory(self) -> sertypes.StateSerializerFactory:
+        # TODO: This factory will probably need to become per-db once
```
Comment (elprans) [inline edb/server/compiler/compiler.py:390]: @msullivan, need your input here.  Config spec can already be different w.r.t enabled extensions, or am I confusing things here?

Comment (msullivan) [reply edb/server/compiler/compiler.py:390]: We don't support doing session configs with extension configs yet, so I think it is fine

Comment (msullivan) [reply edb/server/compiler/compiler.py:390]: Hopefully I will implement that in the next month but I'll try to fix it then

**Review by elprans — APPROVED** [review]

**Pull request merged**